pkg/flow/internal/controller/loader.go

```diff
@@ -56,6 +58,19 @@ func NewLoader(globals ComponentGlobals) *Loader {
 	if globals.Registerer != nil {
 		globals.Registerer.MustRegister(cc)
 	}
+
+	globals.Clusterer.Node.Observe(ckit.FuncObserver(func(peers []peer.Peer) (reregister bool) {
+		for _, cmp := range l.Components() {
+			if globals.Clusterer.IsRegistered(cmp.ID().String()) {
+				err := cmp.Revisit()
+				if err != nil {
+					level.Error(globals.Logger).Log("msg", "failed to revisit component", "componentID", cmp.NodeID(), "err", err)
+				}
+			}
+		}
+		return true
+	}))
```

Comment:
Is this the best place for registering the observer? What other alternatives do we have?

Reply:
Also, this does not easily allow for _other_ types of registrations; here components will only have their Update method called again. Keeping an eye out for other ideas.